File: c/chapitre_2/exo_23.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main()
{
    __pid_t pidFils ;
    int i ;
    
    for (i = 0 ; i < 5 ; i++)
    {
        switch ( pidFils = fork() )
        {
        case -1 :
            perror("Erreur Fork\n") ;
            exit(-1);
            break;
        case 0 : // On est dans le fils faut quitter le boucle
            printf("Je suis le Fils %d\n" , i+1 );
            exit(1);            
            break;
        default :
            if (i == 0)
            printf("Je suis le Père \n");            
            break;
        }
    }
    return 0 ;
}

File: python/chapitre_2/exo23.py
import os
import sys

def main():
    for i in range(5):
        pidFils = os.fork()

        if pidFils == -1:
            print("Erreur Fork")
            sys.exit(-1)
        elif pidFils == 0:
            print(f"Je suis le Fils {i+1}")
            os._exit(1)
        else:
            if i == 0:
                print("Je suis le Père")

if __name__ == "__main__":
    main()